src/com/owncloud/android/utils/DisplayUtils.java

```diff
@@ -207,8 +207,17 @@ else if ((System.currentTimeMillis() - time) < 60 * 1000) {
                 dateString = DateUtils.getRelativeDateTimeString(c, time, minResolution, transitionResolution, flags);
             }
         }
-        
-        return dateString.toString().split(",")[0];
+
+        String[] parts = dateString.toString().split(",");
+        if (parts.length == 2) {
+            if(parts[0].contains(":") && !parts[1].contains(":")) {
+                return parts[1];
+            } else if(parts[1].contains(":") && !parts[0].contains(":")) {
+                return parts[0];
+            }

```

Comment:
The `else if` branch is expected to be executed most of the times. Swapping `if` and `else if` branches would thus bring a tiny performance increase. (It is just two ``contains`` and marginal logic operators less, but this code is executed for *every* file in the current directory...)